### ISSUE TITLE
0.4.3.131 and check_pdh - _value is added to the Performance Data String

### ISSUE THREAD

**Issue by box293** [mickem/nscp#128]
```
Command:
check_nrpe -H 10.25.254.142 -c check_pdh -a 'counter:% Processor Used Core 1=\Processor(0)\% Processor Time' 'warning=value>30' 'critical=value>40'

Output:
OK: % Processor Used Core 1 = 1|'% Processor Used Core 1_value'=1;30;40
```
|'% Processor Used Core 1_value'=

should be

|'% Processor Used Core 1'=

**Comment by mickem** [comment]
define should be ;)

but yes, and this should be fixed in the last build though or atleast I fixed it a while back...

**Comment by mickem** [comment]
please reopen if this is not fixed in the last build (0.4.3.142)

**Comment by box293** [comment]
This does NOT appear to be fixed in 0.4.3.143:

```
Command:
check_nrpe -H win2008r2-01 

Output:
I (0.4.3.143 2015-04-29) seem to be doing fine...

Command:
check_nrpe -H win2008r2-01 -c check_pdh -a 'counter:% Processor Used Core 0=\Processor(0)\% Processor Time' 'warning=value>30' 'critical=value>40'

Output:
OK: % Processor Used Core 0 = 0|'% Processor Used Core 0_value'=0;30;40
```

**Comment by mickem** [comment]
Sorry, my bad... I only fixed it "default" for the CheckCounter alias.

If you want it for check_pdh you can add the perf-config=*(suffix:none) option.

Like so:

```
check_pdh "counter:cpu=\\Processor(0)\\% Processor Time" perf-config=*(suffix:none)
OK: OK:
Performance data: 'cpu'=15;0;0
```

**Comment by box293** [comment]
Thanks for getting back to me. I just tried that however it adds "none" now in place of "_value"

```
Command:
check_nrpe -H win2008r2-01 -c check_pdh -a 'counter:% Processor Used Core 0=\Processor(0)\% Processor Time' 'warning=value>30' 'critical=value>40' perf-config=*(suffix:none)

Output:
OK: % Processor Used Core 0 = 0|'% Processor Used Core 0none'=0;30;40
```

**Comment by mickem** [comment]
Are you sure you have the correct build? 
This is what should be fixed in 0.4.3.143

**Comment by box293** [comment]
I just installed 0.4.3.143 on a clean install of Windows 7 using NSCP-0.4.3.143-x64.msi.

```
Command:
check_nrpe -H 10.25.254.142 

Output:
I (0.4.3.143 2015-04-29) seem to be doing fine...
```

Then did a check_pdh:
```
Command:
check_nrpe -H 10.25.254.142 -c check_pdh -a 'counter:% Processor Used Core 0=\Processor(0)\% Processor Time' 'warning=value>30' 'critical=value>40'

Output:
OK: % Processor Used Core 0 = 0|'% Processor Used Core 0_value'=0;30;40
```

**Comment by mickem** [comment]
Just to clearify in 0.4.3 with check_pdh you will still need perf-config=*(suffix:none) as I dont want to change the current default syntax in a minior release.

In 0.4.3 with CheckCounter you do not need perf-config=*(suffix:none) as it will be added for you.

In 0.4.4 (not released) it will be the default to not use _value (i.e. you wont need perf-config=*(suffix:none)).

So try:
```
check_nrpe -H 10.25.254.142 -c check_pdh -a 'counter:% Processor Used Core 0=\Processor(0)\% Processor Time' 'warning=value>30' 'critical=value>40' perf-config=*(suffix:none)

check_nrpe -H 10.25.254.142 -c check_pdh -a \
    'counter:% Processor Used Core 0=\Processor(0)\% Processor Time' \
    'warning=value>30' \
    'critical=value>40' \
    perf-config=*(suffix:none)
```

**Comment by box293** [comment]
Ahh don't I feel like an idiot, I didn't test with the suffix bit, sorry about that.

HOWEVER by doing that, "_value" is now "none"

```
Command:
check_nrpe -H 10.25.254.142 -c check_pdh -a 'counter:% Processor Used Core 0=\Processor(0)\% Processor Time' 'warning=value>30' 'critical=value>40' perf-config=*(suffix:none)

Output:
OK: % Processor Used Core 0 = 1|'% Processor Used Core 0none'=1;30;40
```

**Comment by box293** [comment]
I've been doing some testing and it appears as though this 'none' stuff (displaying the word none) has been fixed in 0.5.0.17. I tested with 0.4.4.2 and it still adds the none word.

**Comment by mickem** [comment]
Ahh, sorry, 0.4.4 has since become 0.5.0 but I guess I could maybe do it for 0.4.4 as well?

**Comment by box293** [comment]
I won't bother adding it to past versions, it people want that fixed they can just upgrade to 0.5.x :)

**Comment by mickem** [comment]
ok, great...
